corehq/apps/ota/views.py

```diff
@@ -92,15 +95,44 @@ def search(request, domain):
         _soft_assert(False, "Someone in domain: {} tried accessing case search without a config".format(domain), e)
         config = CaseSearchConfig(domain=domain)
 
+    query_addition_id = criteria.pop(SEARCH_QUERY_ADDITION_KEY, None)
+
     fuzzies = config.config.get_fuzzy_properties_for_case_type(case_type)
     for key, value in criteria.items():
         search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
-    results = search_es.values()
+
+    query_addition_debug_details = {}
+    _add_case_search_addition(request, search_es, query_addition_id, query_addition_debug_details)
+
+    try:
+        results = search_es.values()
+    except Exception as e:
+        notify_exception(request, e.message, details=dict(
+            exception_type=type(e),
+            **query_addition_debug_details
+        ))
+        raise
     # Even if it's a SQL domain, we just need to render the results as cases, so CommCareCase.wrap will be fine
     cases = [CommCareCase.wrap(flatten_result(result)) for result in results]
     fixtures = CaseDBFixture(cases).fixture
     return HttpResponse(fixtures, content_type="text/xml")
 
+def _add_case_search_addition(request, search_es, query_addition_id, query_addition_debug_details):
+    if query_addition_id:
+        query_addition = CaseSearchQueryAddition.objects.get(id=query_addition_id)
```

Comment:
Not sure what the behaviour should be if this object doesn't exist... but this will fail hard I think. That might be right though, since it will send an error email.